packages/docregistry/src/context.ts

```diff
@@ -219,6 +224,14 @@ export class Context<
     return this._readyPromise;
   }
 
+  /**
+   * Returns whether this document context is writable. True only if the
+   * contents model is writable and collaboration is disabled.
+   */
+  get writable(): boolean {
+    return !!(this._contentsModel?.writable && !this._collaborative);
+  }
```

Comment:
The problem is that `_contentsModel` is of type `Contents.IModel`, while `_model` is of type `DocumentRegistry.IModel`. So because only the DocumentContext has access to both of these models, it should be the single source of truth for whether this document is writable, which is why I think it makes sense to keep this property here.